desafio-06/AlexPHorta/python/desafio06.py

```diff
@@ -0,0 +1,326 @@
+        le = sieve_less_or_equal(expression, w)
+        sw = sieve_starts_with(expression, w)
+        rm = sieve_remaining(expression, w)
+        ql = sieve_number_of_letters(expression, w)
+        if all((le, sw, rm, ql)):
+            res.append(w)
+    return res
+
+def sieve_number_of_letters(expression, word):
+    """Check if a word is contained in expression. Remove those that can't be.
+
+    Positional arguments
+    expression -- the expression to be used for the search of anagrams
+    word -- the word to be compared
+    """
+    expression_quant = quant_letters(expression)
+    word_quant = quant_letters(word)
+    for k, v in word_quant.items():
+        if expression_quant.get(k) is not None and v > expression_quant[k]:
+            return False
+    return True
+
+def quant_letters(a_word):
+    """Return a mapping of the number of the different letters in a word.
+
+    Positional arguments
+    a_word -- the word to be mapped
+    """
+    keys = set(a_word)
+    quanto = {k:a_word.count(k) for k in keys}
+    return quanto
+
+def sieve_remaining(expression, word):
+    """Remove any word that has letters that are not in expression.
+
+    Positional arguments
+    expression -- the expression to be used for the search of anagrams
+    word -- the word to be evaluated
+    """
+    letters = set(expression)
+    uppercase = set(string.ascii_uppercase)
+    remaining = uppercase - letters
+    return not any((l in remaining) for l in word)
+
+def sieve_starts_with(expression, word):
+    """Return only words that begin with one of the letters in expression.
+
+    Positional arguments
+    expression -- the expression to be used for the search of anagrams
+    word -- the word to be evaluated
+    """
+    letters = set(expression)
+    return any(word.startswith(l) for l in letters)
+
+def sieve_less_or_equal(expression, word):
+    """Exclude words that are lengthier than expression.
+
+    Positional arguments
+    expression -- the expression to be used for the search of anagrams
+    word -- the word to be evaluated
+    """
+    return len(expression) >= len(word)
+
+# Peguei de https://jeromekelleher.net/generating-integer-partitions.html
+def accel_asc(n):
+    """Yield all partitions of a given integer.
+    E.g.: The partitions of three are: 1,1,1; 1,2; 2,1; 3
+
+    Positional arguments
+    n -- the integer to be used
+    """
+    a = [0 for i in range(n + 1)]
+    k = 1
+    y = n - 1
+    while k != 0:
+        x = a[k - 1] + 1
+        k -= 1
+        while 2 * x <= y:
+            a[k] = x
+            y -= x
+            k += 1
+        l = k + 1
+        while x <= y:
+            a[k] = x
+            a[l] = y
+            yield a[:k + 2]
+            x += 1
+            y -= 1
+        a[k] = x + y
+        y = x + y - 1
+        yield a[:k + 1]
+
+
+if __name__ == "__main__":
+
+    import gettext
+    import sys
+
+    def translate(text):
+        text = text.replace("usage", "modo de usar")
```

Comment:
Dá erro se eu digito números como argumento.
Você tem que tratar isso:
➜  testes-desafios git:(master) ✗ python3 des6.py 902823
Traceback (most recent call last):
  File "/home/marcelo/testes-desafios/des6.py", line 318, in <module>
    if len(args.termo) == 0:
           ^^^^^^^^^^